testutils/script.go

```diff
@@ -0,0 +1,499 @@
+	db := getDB(ts)
+	txn := db.ReadTxn()
+	tbls := db.GetTables(txn)
+	var buf bytes.Buffer
+	w := tabwriter.NewWriter(&buf, 5, 4, 3, ' ', 0)
+	fmt.Fprintf(w, "Name\tObject count\tIndexes\n")
+	for _, tbl := range tbls {
+		idxs := strings.Join(tbl.Indexes(), ", ")
+		fmt.Fprintf(w, "%s\t%d\t%s\n", tbl.Name(), tbl.NumObjects(txn), idxs)
+	}
+	w.Flush()
+	ts.Logf("%s", buf.String())
+}
+
+func ShowTableCmd(ts *testscript.TestScript, neg bool, args []string) {
+	db := getDB(ts)
+	if len(args) != 1 {
+		ts.Fatalf("usage: show_table <table>")
+	}
+	ts.Logf("%s", showTable(ts, db, args[0]).String())
+}
+
+func WriteTableCmd(ts *testscript.TestScript, neg bool, args []string) {
+	if len(args) < 1 || len(args) > 5 {
+		ts.Fatalf("usage: write_table <table> (-to=<file>) (-columns=<col1,col2,...>) (-format={table*,yaml})")
+	}
+	var flags flag.FlagSet
+	file := flags.String("to", "", "File to write to instead of stdout")
+	columns := flags.String("columns", "", "Comma-separated list of columns to write")
+	format := flags.String("format", "table", "Format to write in")
+
+	// Sort the args to allow the table name at any position.
+	slices.SortFunc(args, func(a, b string) int {
+		switch {
+		case a[0] == '-':
+			return 1
+		case b[0] == '-':
+			return -1
+		default:
+			return cmp.Compare(a, b)
+		}
+	})
+	err := flags.Parse(args[1:])
+	if err != nil {
+		ts.Fatalf("bad args: %s", err)
+	}
+	tableName := args[0]
+
+	db := getDB(ts)
+
+	switch *format {
+	case "yaml", "json":
+		if len(*columns) > 0 {
+			ts.Fatalf("-columns not supported with -format=yaml/json")
+		}
+
+		txn := db.ReadTxn()
+		meta := db.GetTable(txn, tableName)
+		if meta == nil {
+			ts.Fatalf("table %q not found", tableName)
+		}
+		tbl := statedb.AnyTable{Meta: meta}
+
+		var buf bytes.Buffer
+		count := tbl.Meta.NumObjects(txn)
+		for obj := range tbl.All(txn) {
+			if *format == "yaml" {
+				out, err := yaml.Marshal(obj)
+				if err != nil {
+					ts.Fatalf("yaml.Marshal: %s", err)
+				}
+				buf.Write(out)
+				if count > 1 {
+					buf.WriteString("---\n")
+				}
+			} else {
+				out, err := json.Marshal(obj)
+				if err != nil {
+					ts.Fatalf("json.Marshal: %s", err)
+				}
+				buf.Write(out)
+				buf.WriteByte('\n')
+			}
+			count--
+		}
+		if *file == "" {
+			ts.Logf("%s", buf.String())
+		} else if err := os.WriteFile(ts.MkAbs(*file), buf.Bytes(), 0644); err != nil {
+			ts.Fatalf("WriteFile(%s): %s", *file, err)
+		}
+	default:
+		var cols []string
+		if len(*columns) > 0 {
+			cols = strings.Split(*columns, ",")
+		}
+		buf := showTable(ts, db, tableName, cols...)
+		if *file == "" {
+			ts.Logf("%s", buf.String())
+		} else if err := os.WriteFile(ts.MkAbs(*file), buf.Bytes(), 0644); err != nil {
+			ts.Fatalf("WriteFile(%s): %s", *file, err)
+		}
+	}
+}
+
+func CompareTableCmd(ts *testscript.TestScript, neg bool, args []string) {
+	var flags flag.FlagSet
+	wait := flags.Duration("wait", 0, "Amount of time to wait for the table contents to match")
```

Comment:
Don't like naming this `-wait`. `-retry`? `-eventually` (too long?)?

Reply:
I don't think we care about exponential backoff in tests that much. I'll make the default that it waits some amount of time and make this `-timeout`.